**Material**

https://docs.google.com/spreadsheets/d/1gU-Y_meE7A-re4LCA3Bq24K7b7vs9U3ZFXy7oS6ftdQ/edit?usp=sharing



In [ ]:
import pandas as pd


In [ ]:
def import_data():
  # Function to retrieve the data and refresh if necessary
  link = "https://docs.google.com/spreadsheets/d/e/2PACX-1vRpFclgUNhF1CSHDHPyQCaSUojcPALed14HFzCMFiseOe1izFPyWOXeDgRABKLhH0ckTW3ffFa1xjRv/pub?output=xlsx"
  df = pd.read_excel(link)
  return df

In [ ]:
df = import_data()
df

English Original Force Dynamics (English)  \
0           A fine day like this made her impatient.                day > she   
1  In high summer, my sister an I were not allowe...      my sister and I < Ø   
2                         She was forbidden to help.                  she < Ø   
3  As though she were the one who was imposing un...             she > fellow   
4           He let the greatcoat fall on the ground.           he > greatcoat   
5  The accumulated inactivity of the summer weeks...         inactivity > she   
6  Being at odds with her father about anything a...      being at odds > she   
7  Emily let her thoughts move away from her elde...         Emily > thoughts   
8        Perhaps I should have written Leon a story.                    I < Ø   
9           I suppose we should start by reading it.                   we < Ø   

                             Finnish Human Reference Force Dynamics (Finnish)  \
0    Tällainen upea päivä sai hänet kärsimättömäksi.                 Retained   
1                        Emme päässeet kesällä ulos.                 Retained   
2                              Auttaa hän ei saanut.                 Retained   
3  Ikään kuin juuri Briony olisiki ollut se, joka...                 Retained   
4           Hän päästi manttelinsa putoamaan maahan.                 Retained   
5  Kai hän kiiruhti askeliaan siksikin, että oli ...                 Modified   
6  Hänestä tuntui kovin epämukavalta joutua ylipä...             Not retained   
7  Emily antoi ajatustensa siirtyä pois vanhemmas...                 Retained   
8              Miksen kirjoittanut Leonille tarinaa?             Not retained   
9                               Luetaan se näytelmä.             Not retained   

                       Finnish MT (Google Translate)  \
0   Tällainen hieno päivä sai hänet kärsimättömäksi.   
1  Keskikesällä siskoni ja minua ei päästetty ulo...   
2                      Häntä kiellettiin auttamasta.   
3  Ikään kuin hän olisi se, joka asetti epämielly...   
4                     Hän antoi takin pudota maahan.   
5  Myös finaalien jälkeen kertynyt kesäviikkojen ...   
6  Ristiriidat isänsä kanssa ylipäänsä mistä taha...   
7  Emily antoi ajatustensa siirtyä pois vanhimmas...   
8  Ehkä minun olisi pitänyt kirjoittaa Leonille t...   
9  Luulen, että meidän pitäisi aloittaa lukemalla...   

  Force Dynamics (Google Translate)  \
0                          Retained   
1                          Retained   
2                          Retained   
3                          Retained   
4                          Retained   
5                          Retained   
6                          Retained   
7                          Retained   
8                          Retained   
9                          Retained   

                              Polish Human Reference Force Dynamics (Polish)  \
0  Taki ładny dzień jak ten powodował, że traciła...                Retained   
1  Latem nie wypuszczali z domu mojej siostry i m...                Modified   
2                           Nie wolno było mu pomóc.                Modified   
3           Jakby to ona poddała go ciężkiej próbie.                Retained   
4                           Rzucił na ziemię szynel.            Not retained   
5  Jej krok przyspieszyła również przedłużająca s...                Modified   
6  Jakakolwiek różnica zdań z ojcem wprawiała ją ...                Retained   
7        Emily przestała rozmyślać o starszej córce.            Not retained   
8       Może powinnam napisać opowiadanie dla Leona?                Retained   
9                     Najpierw trzeba ją przeczytać.                Modified   

                        Polish MT (Google Translate)  \
0                Taki piękny dzień niecierpliwił ją.   
1  W środku lata mojej siostrze i mnie nie wolno ...   
2                               Zakazano jej pomocy.   
3  Jakby to ona poddawała bezbronnemu chłopczykow...   
4                         Upuścił płaszcz na

In [ ]:
!pip install openai==0.28

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 76.5/76.5 kB 6.0 MB/s eta 0:00:00
  Attempting uninstall: openai
    Found existing installation: openai 1.57.4
    Uninstalling openai-1.57.4:
      Successfully uninstalled openai-1.57.4


In [ ]:
import os
import openai
openai.api_key = os.getenv("OPENAI_API_KEY")


In [ ]:
def get_response(user_content, system_content, model):
    completion =  openai.ChatCompletion.create(
    model= model, #"gpt-4-0613",#"gpt-3.5-turbo-0613", gpt-4o
    temperature=0,
    messages=[
    {"role": "system", "content": system_content},
    {"role": "user", "content": user_content}
  ]
)
    return(completion.choices[0].message.content)

In [ ]:
# Ben proposal 21.2.2024
# source_language = 'English'
# target_language = 'Polish'
# system_prompt = f'''You are assistant translator. You know to evaluate translations.'''
# instruction_prompt = f'''Provide a translation of text from {source_language} to {target_language} and then evaluate the quality of two translations
#  1) you will evaluate your translation as translation_quality_GPT
#  2) you will evaluate user translation as translation_quality_user

# The quality should be qualitative as well as numerical as a float from  0 to 1.
# The output should be in the JSON.
# Do not provide nothing but JSON.

# example output with keys:'''+\
# '''{'original_text': str, 'translation_GPT':str, 'translation_user':str,
# 'trans_quality_GPT_q': str,
# 'trans_quality_GPT_n': float,
# 'trans_quality_user_q': string,
# 'trans_quality_user_n': float}

# original_text:
# '''
# original_text = 'A fine day like this made her impatient.'
# translation_user = 'Taki ładny dzień jak ten powodował, że traciła cierpliwość.'


**Sample 1:** *A fine day like this **made** her **impatient**.*



In [ ]:
# English - Finnish (Human Reference)
source_language = 'English'
target_language = 'Finnish'
original_text = 'A fine day like this made her impatient.'
translation_user = 'Tällainen upea päivä sai hänet kärsimättömäksi.'

system_prompt = f'''You are an expert linguist and translator. You know how to evaluate translations.'''

instruction_prompt = f'''Provide a translation of a given sentence from {source_language} to {target_language}.
Describe your translation, as well as the user reference provided, with an exclusive focus on their verb phrases.
Describe the verb phrases in terms of lexis, syntax, and semantics.
Evaluate the quality of the translations of the verb phrases, taking into account lexis, syntax, and semantics.
With lexis, and syntax regarding the verb phrases, provide as many technical details as possible.
When describing the semantics of the verb phrases, take into account Talmy's notion of Force Dynamics.
Apart from the word choices and grammar solutions around the verb phrases, do not forget about language conventions and idiomacy of both outputs.

 1) You will describe YOUR output as translation_description_GPT.
 2) You will evaluate YOUR output as translation_quality_GPT.
 3) You will describe USER output as translation_description_user.
 4) You will evaluate USER output as translation_quality_user.

The description of the verb phrase should be qualitative in terms of lexis, syntax, and semantics, and the evaluation of the verb phrase should be numerical as a float from  0 to 1.
The output should be in the JSON.
Do not provide nothing but JSON.

example output with keys:'''+\
'''{'original_text': str, 'translation_GPT':str, 'translation_user':str,
'trans_description_GPT_q': str,
'trans_quality_GPT_n': float,
'trans_description_user_q': string,
'trans_quality_user_n': float}

original_text:\n
'''


In [ ]:
source_language = 'English'
target_language = 'Finnish'
original_text = 'A fine day like this made her impatient.'
translation_user = 'Tällainen upea päivä sai hänet kärsimättömäksi.'

In [ ]:
print(get_response(instruction_prompt+original_text+'\ntranslation_user:\n'+translation_user, system_prompt, 'gpt-4o'))

{
    "original_text": "A fine day like this made her impatient.",
    "translation_GPT": "Tällainen hieno päivä teki hänet kärsimättömäksi.",
    "translation_user": "Tällainen upea päivä sai hänet kärsimättömäksi.",
    "trans_description_GPT_q": "The verb phrase 'made her impatient' is translated as 'teki hänet kärsimättömäksi'. Lexically, 'made' is translated to 'teki', which is a direct translation of the verb 'to make' in Finnish. Syntactically, the structure follows the Finnish pattern of subject-verb-object, with 'hänet' (her) as the object and 'kärsimättömäksi' (impatient) as the complement. Semantically, the verb 'teki' implies a causative action, aligning with Talmy's Force Dynamics where the day exerts force to change her state to impatient.",
    "trans_quality_GPT_n": 0.9,
    "trans_description_user_q": "The verb phrase 'made her impatient' is translated as 'sai hänet kärsimättömäksi'. Lexically, 'sai' translates to 'got' or 'caused', which is a slightly different choice

In [ ]:
system_prompt = f'''You are an expert linguist and translator. You know how to evaluate translations.'''
# English - Finnish (Google Translate)
def translation_evaluation(source_language, target_language, original_text, translation_Google, translation_User):
  # source_language = 'English'
  # target_language = 'Finnish'
  # original_text = 'A fine day like this made her impatient.'
  # translation_Google = 'Tällainen hieno päivä sai hänet kärsimättömäksi.'



  instruction_prompt = f'''Provide a translation of a given sentence from {source_language} to {target_language}.
  Describe your translation, as well as the Google Translate reference provided, with an exclusive focus on their verb phrases.
  Describe the verb phrases in terms of lexis, syntax, and semantics.
  Evaluate the quality of the translations of the verb phrases, taking into account lexis, syntax, and semantics.
  With lexis, and syntax regarding the verb phrases, provide as many technical details as possible.
  When describing the semantics of the verb phrases, take into account Talmy's notion of Force Dynamics.
  Apart from the word choices and grammar solutions around the verb phrases, do not forget about language conventions and idiomacy of both outputs.
  Compare the quality scores of: 1) YOUR output, 2)  GOOGLE output, and 3) HUMAN output. Describe in detail what were criteria for that particular scores. Also, describe in comparative terms only the verb phrases.

  1) You will describe YOUR output as translation_description_GPT. Describe only the translation of the verb phrase.
  2) You will evaluate YOUR output as translation_quality_GPT. Evaluate only the translation of the verb phrase.
  3) You will describe GOOGLE output as translation_description_Google. Describe only the translation of the verb phrase.
  4) You will evaluate GOOGLE output as translation_quality_Google. Evaluate only the translation of the verb phrase.
  5) You will describe HUMAN output as translation_description_User. Describe only the translation of the verb phrase.
  6) You will evaluate HUMAN output as translation_quality_User. Evaluate only the translation of the verb phrase.
  7) You will compare quality scores of YOUR output, GOOGLE output, and HUMAN output. Describe what you based your score on. Base your comparison only on the verb phrases.

  The description of the verb phrase should be qualitative in terms of lexis, syntax, and semantics, and the evaluation of the verb phrase should be numerical as a float from  0 to 1.
  The output should be in the JSON!

  example output with keys:'''+\
  '''{'original_text': str, 'translation_GPT':str, 'translation_Google':str, 'translation_User':str,
  'trans_description_GPT_q': str,
  'trans_quality_GPT_n': float,
  'trans_description_Google_q': string,
  'trans_quality_Google_n': float,
  'trans_description_User_q': string,
  'trans_quality_User_n': float,
  'trans_quality_compare': string

  }

  Do not provide nothing but JSON!
  \n\n original_text:
  '''
  return (instruction_prompt+original_text+'\ntranslation_Google:\n'+translation_Google+'\ntranslation_User:\n'+translation_User)

instruct_evaluation = translation_evaluation('English','Finnish', 'A fine day like this made her impatient.', 'Tällainen hieno päivä sai hänet kärsimättömäksi.', 'My awful translation')
instruct_evaluation

"Provide a translation of a given sentence from English to Finnish.\n  Describe your translation, as well as the Google Translate reference provided, with an exclusive focus on their verb phrases.\n  Describe the verb phrases in terms of lexis, syntax, and semantics.\n  Evaluate the quality of the translations of the verb phrases, taking into account lexis, syntax, and semantics.\n  With lexis, and syntax regarding the verb phrases, provide as many technical details as possible.\n  When describing the semantics of the verb phrases, take into account Talmy's notion of Force Dynamics.\n  Apart from the word choices and grammar solutions around the verb phrases, do not forget about language conventions and idiomacy of both outputs.\n  Compare the quality scores of: 1) YOUR output, 2)  GOOGLE output, and 3) HUMAN output. Describe in detail what were criteria for that particular scores. Also, describe in comparative terms only the verb phrases.\n\n  1) You will describe YOUR output as trans

In [ ]:
df.columns

Index(['English Original', 'Force Dynamics (English)',
       'Finnish Human Reference', 'Force Dynamics (Finnish)',
       'Finnish MT (Google Translate)', 'Force Dynamics (Google Translate)',
       'Polish Human Reference', 'Force Dynamics (Polish)',
       'Polish MT (Google Translate)', 'Force Dynamics (Google Translate).1'],
      dtype='object')

In [ ]:
import json
def responses_dictionary(df,
                        source_language,
                        target_language,
                        original_text_row,
                        translation_Google_row,
                        translation_User_row ):
  dict_responses = []
  df_results = pd.DataFrame()
  for index, row in df[0:].iterrows():
    original_text = row[original_text_row] # row['English Original']
    translation_Google = row[translation_Google_row] # row['Finnish MT (Google Translate)']
    translation_User = row[translation_User_row]
    instruction = translation_evaluation(source_language, target_language, original_text, translation_Google, translation_User)

    dict_response = get_response(instruction, system_prompt, 'gpt-4o')
    dict_responses.append(dict_response.replace('```json','').replace('```',''))
    try:
      dicts = [json.loads(d) for d in dict_responses]
      df_results = pd.DataFrame(dicts)
    except:
      df_results = dict_responses

  return df_results

# Example of use
# responses_dictionary('English','Finnish', 'English Original', 'Finnish MT (Google Translate)', 'Finnish Human Reference')



## Finnish

In [ ]:
dict_responses_Finnish = responses_dictionary(df, 'English','Finnish', 'English Original', 'Finnish MT (Google Translate)', 'Finnish Human Reference')
dict_responses_Finnish

original_text  \
0           A fine day like this made her impatient.   
1  In high summer, my sister and I were not allow...   
2                         She was forbidden to help.   
3  As though she were the one who was imposing un...   
4           He let the greatcoat fall on the ground.   
5  The accumulated inactivity of the summer weeks...   
6  Being at odds with her father about anything a...   
7  Emily let her thoughts move away from her elde...   
8        Perhaps I should have written Leon a story.   
9           I suppose we should start by reading it.   

                                     translation_GPT  \
0  Tällainen hieno päivä teki hänet kärsimättömäksi.   
1  Keskikesällä siskoni ja minä emme saaneet menn...   
2                    Hänelle kiellettiin auttaminen.   
3  Ikään kuin hän olisi se, joka pakotti epämiell...   
4              Hän antoi päällystakin pudota maahan.   
5  Kesäviikkojen kertynyt toimettomuus finaalien ...   
6  Hänen oli epämukavaa olla eri mieltä isänsä ka...   
7  Emily antoi ajatustensa siirtyä pois vanhimmas...   
8  Ehkä minun olisi pitänyt kirjoittaa Leonille t...   
9  Luulen, että meidän pitäisi aloittaa lukemalla...   

                                  translation_Google  \
0   Tällainen hieno päivä sai hänet kärsimättömäksi.   
1  Keskikesällä siskoni ja minua ei päästetty ulo...   
2                      Häntä kiellettiin auttamasta.   
3  Ikään kuin hän olisi se, joka asetti epämielly...   
4                     Hän antoi takin pudota maahan.   
5  Myös finaalien jälkeen kertynyt kesäviikkojen ...   
6  Ristiriidat isänsä kanssa ylipäänsä mistä taha...   
7  Emily antoi ajatustensa siirtyä pois vanhimmas...   
8  Ehkä minun olisi pitänyt kirjoittaa Leonille t...   
9  Luulen, että meidän pitäisi aloittaa lukemalla...   

                                    translation_User  \
0    Tällainen upea päivä sai hänet kärsimättömäksi.   
1                        Emme päässeet kesällä ulos.   
2                              Auttaa hän ei saanut.   
3  Ikään kuin juuri Briony olisiki ollut se, joka...   
4           Hän päästi manttelinsa putoamaan maahan.   
5  Kai hän kiiruhti askeliaan siksikin, että oli ...   
6  Hänestä tuntui kovin epämukavalta joutua ylipä...   
7  Emily antoi ajatustensa siirtyä pois vanhemmas...   
8              Miksen kirjoittanut Leonille tarinaa?   
9                               Luetaan se näytelmä.   

                             trans_description_GPT_q  trans_quality_GPT_n  \
0  The verb phrase 'teki hänet kärsimättömäksi' u...                 0.90   
1  The verb phrase 'emme saaneet mennä' uses the ...                 0.90   
2  The verb phrase 'kiellettiin auttaminen' uses ...                 0.85   
3  The verb phrase 'pakotti' is a past tense form...                 0.90   
4  The verb phrase 'antoi pudota' uses the verb '...                 0.90   
5  The verb phrase 'kiirehti häntä eteenpäin' use...                 0.85   
6  The verb phrase 'olla eri mieltä' (to be of a ...                 0.85   
7  The verb phrase 'let her thoughts move away' i...                 0.95   
8  The verb phrase 'should have written' is trans...                 0.90   
9  The verb phrase 'should start by reading' is t...                 0.90   

                          trans_description_Google_q  trans_quality_Google_n  \
0  The verb phrase 'sai hänet kärsimättömäksi' us...                    0.80   
1  The verb phrase 'ei päästetty' uses the passiv...                    0.85   
2  The verb phrase 'kiellettiin auttamasta' uses ...                    0.90   
3  The verb phrase 'asetti' is a past tense form ...                    0.70   
4  The verb phrase 'antoi pudota' uses 'antaa' (t...                    0.85   
5  The verb phrase 'kiihdytti häntä mukaan' uses ...                    0.75   
6  The verb phrase 'sai hänet epämukavaksi' (made...                    0.75   
7  The verb phrase 'let her thoughts move away' i...                    0.95   
8  

In [ ]:
dict_responses_Finnish.to_excel('Finnish_evaluation.xlsx')

## Polish

In [ ]:
dict_responses_Polish = responses_dictionary(df, 'English','Polish', 'English Original', 'Polish MT (Google Translate)', 'Polish Human Reference')
dict_responses_Polish

original_text  \
0           A fine day like this made her impatient.   
1  In high summer, my sister and I were not allow...   
2                         She was forbidden to help.   
3  As though she were the one who was imposing un...   
4           He let the greatcoat fall on the ground.   
5  The accumulated inactivity of the summer weeks...   
6  Being at odds with her father about anything a...   
7  Emily let her thoughts move away from her elde...   
8        Perhaps I should have written Leon a story.   
9           I suppose we should start by reading it.   

                                     translation_GPT  \
0   Taki piękny dzień sprawił, że była niecierpliwa.   
1  W środku lata nie pozwalano mojej siostrze i m...   
2                            Zabroniono jej pomagać.   
3  Jakby to ona narzucała bezbronnemu chłopcu nie...   
4               Pozwolił płaszczowi upaść na ziemię.   
5  Nagromadzona bezczynność letnich tygodni po fi...   
6  Bycie w konflikcie z ojcem w jakiejkolwiek spr...   
7  Emily pozwoliła swoim myślom oddalić się od na...   
8    Może powinnam była napisać Leonowi opowiadanie.   
9  Sądzę, że powinniśmy zacząć od przeczytania tego.   

                                  translation_Google  \
0                Taki piękny dzień niecierpliwił ją.   
1  W środku lata mojej siostrze i mnie nie wolno ...   
2                               Zakazano jej pomocy.   
3  Jakby to ona poddawała bezbronnemu chłopczykow...   
4                         Upuścił płaszcz na ziemię.   
5  Skumulowana bezczynność podczas letnich tygodn...   
6  To, że nie zgadzała się z ojcem w jakiejkolwie...   
7  Emily pozwoliła myślom odejść od najstarszej c...   
8         Może powinnam napisać Leonowi opowiadanie.   
9       Myślę, że powinniśmy zacząć od przeczytania.   

                                    translation_User  \
0  Taki ładny dzień jak ten powodował, że traciła...   
1  Latem nie wypuszczali z domu mojej siostry i m...   
2                           Nie wolno było mu pomóc.   
3           Jakby to ona poddała go ciężkiej próbie.   
4                           Rzucił na ziemię szynel.   
5  Jej krok przyspieszyła również przedłużająca s...   
6  Jakakolwiek różnica zdań z ojcem wprawiała ją ...   
7        Emily przestała rozmyślać o starszej córce.   
8       Może powinnam napisać opowiadanie dla Leona?   
9                     Najpierw trzeba ją przeczytać.   

                             trans_description_GPT_q  trans_quality_GPT_n  \
0  The verb phrase 'sprawił, że była niecierpliwa...                  0.9   
1  The verb phrase 'nie pozwalano' is a passive c...                  0.9   
2  The verb phrase 'was forbidden to help' is tra...                  0.9   
3  The verb phrase 'narzucała' is a past imperfec...                  0.9   
4  The verb phrase 'Pozwolił płaszczowi upaść' us...                  0.9   
5  The verb phrase 'przyspieszała' is a past tens...                  0.9   
6  The verb phrase 'sprawiało, że czuła się niesw...                  0.9   
7  The verb phrase 'pozwoliła swoim myślom oddali...                  0.9   
8  The verb phrase 'should have written' is trans...                  0.9   
9  The verb phrase 'powinniśmy zacząć' uses the m...                  0.9   

                          trans_description_Google_q  trans_quality_Google_n  \
0  The verb phrase 'niecierpliwił ją' uses 'nieci...                    0.80   
1  The verb phrase 'nie wolno było' is a passive ...                    0.85   
2  The verb phrase 'was forbidden to help' is tra...                    0.80   
3  The verb phrase 'poddawała' is a past imperfec...                    0.80   
4  The verb phrase 'Upuścił płaszcz' uses 'upuści...                    0.70   
5  The verb phrase 'przyspieszała' is used simila...                    0.90   
6  The verb phrase 'sprawiało, że czuła się nieko...                    0.85   
7  The verb phrase 'pozwoliła myślom odejść' uses...                    0.85   
8  

In [ ]:
dict_responses_Polish.to_excel('Polish_evaluation.xlsx')

# Croatian

In [ ]:
link_hr = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vSCb5y3L7siAReDJ2QgaJCoR2b_GnuEciIoGRfdzigi1J0-7IJuSTc2K3WlmH-87czMtKNQsB0oUmAj/pub?output=xlsx'
df = pd.read_excel(link_hr)
df

Unnamed: 0                                   English Original  \
0           1           A fine day like this made her impatient.   
1           2  In high summer, my sister and I were not allow...   
2           3                         She was forbidden to help.   
3           4  As though she were the one who was imposing un...   
4           5           He let the greatcoat fall on the ground.   
5           6  The accumulated inactivity of the summer weeks...   
6           7  Being at odds with her father about anything a...   
7           8  Emily let her thoughts move away from her elde...   
8           9        Perhaps I should have written Leon a story.   
9          10           I suppose we should start by reading it.   

        Force Dynamics                                    GPT Translation  \
0            day > she    Lijep dan poput ovog učinio ju je nestrpljivom.   
1  my sister and I < Ø  U visokom ljetu, sestra i ja nismo smjele izać...   
2              she < Ø                     Bilo joj je zabranjeno pomoći.   
3         she > fellow  Kao da je ona bila ta koja je nametala neugodn...   
4       he > greatcoat                   Pustio je kaput da padne na tlo.   
5     inactivity > she  Nagomilana neaktivnost ljetnih tjedana od fina...   
6  being at odds > she  Bilo joj je neugodno biti u sukobu s ocem oko ...   
7     Emily > thoughts  Emily je dopustila da joj misli odlutaju od na...   
8                I < Ø             Možda sam trebao napisati Leonu priču.   
9               we < Ø  Pretpostavljam da bismo trebali početi čitajuć...   

  Force Dynamics.1                     Croatian MT (Google Translate)  \
0         Retained   Lijep dan poput ovoga učinio ju je nestrpljivom.   
1         Retained  U jeku ljeta sestra i ja nismo smjele izlaziti...   
2         Retained                   Bilo joj je zabranjeno pomagati.   
3         Retained  Kao da je ona ta koja je bespomoćnom mališanu ...   
4         Retained                   Pustio je kaput da padne na tlo.   
5         Retained  Nagomilana neaktivnost tijekom ljetnih tjedana...   
6     Not retained  Bilo joj je neugodno to što je bila u svađi s ...   
7         Retained  Emily je dopustila da joj se misli odmaknu od ...   
8         Retained             Možda sam trebao Leonu napisati priču.   
9         Retained  Pretpostavljam da bismo trebali početi s čitan...   

  Force Dynamics.2                           Croatian Human Reference  \
0         Retained   Lijep dan poput ovoga učinio ju je nestrpljivom.   
1         Retained  Usred ljeta sestra i ja nismo smjele izlaziti ...   
2         Retained                        Zabranili su joj da pomaže.   
3         Retained  Kao da je ona bila ta koja je bespomoćnom mali...   
4         Retained                   Pustio je kaput da padne na tlo.   
5         Retained  Nagomilana neaktivnost tijekom ljetnih tjedana...   
6     Not retained  Bilo joj je neugodno to što je bila u svađi s ...   
7         Retained  Emily je dopustila da joj misli odmaknu od naj...   
8         Retained       Možda sam trebala napisati Leonu neku priču.   
9         Retained  Pretpostavljam da bismo trebali početi s čitan...   

     Force Dynamics.3                        GPT Translation Description  \
0            Retained  The verb phrase 'učinio ju je nestrpljivom' co...   
1            Retained  The verb phrase 'nismo smjele izaći' uses the ...   
2  Retained/Modified?  The verb phrase 'bilo joj je zabranjeno pomoći...   
3            Retained  The verb phrase 'was imposing' is translated a...   
4            Retained  The verb phrase 'let fall' is translated as 'p...   
5            Retained  The verb phrase 'požurivala' is a past tense f...   
6        Not retained  The verb phrase 'biti u sukobu' (to be in conf...   
7            Retained  The verb phrase 'dopustila da joj misli odluta...   
8            Retained  The verb phrase 'should have written' is trans...   
9            Retained  The verb phrase

In [ ]:
dict_responses_Croatian = responses_dictionary(df, 'English','Croatian', 'English Original', 'Croatian MT (Google Translate)', 'Croatian Human Reference')
dict_responses_Croatian

original_text  \
0           A fine day like this made her impatient.   
1  In high summer, my sister and I were not allow...   
2                         She was forbidden to help.   
3  As though she were the one who was imposing un...   
4           He let the greatcoat fall on the ground.   
5  The accumulated inactivity of the summer weeks...   
6  Being at odds with her father about anything a...   
7  Emily let her thoughts move away from her elde...   
8        Perhaps I should have written Leon a story.   
9           I suppose we should start by reading it.   

                                     translation_GPT  \
0    Lijep dan poput ovog učinio ju je nestrpljivom.   
1  U visokom ljetu, sestra i ja nismo smjele izać...   
2                     Bilo joj je zabranjeno pomoći.   
3  Kao da je ona bila ta koja je nametala neugodn...   
4                   Pustio je kaput da padne na tlo.   
5  Nagomilana neaktivnost ljetnih tjedana od fina...   
6  Bilo joj je neugodno biti u sukobu s ocem oko ...   
7  Emily je dopustila da joj misli odlutaju od na...   
8            Možda sam trebala napisati Leonu priču.   
9  Pretpostavljam da bismo trebali početi čitajuć...   

                                  translation_Google  \
0   Lijep dan poput ovoga učinio ju je nestrpljivom.   
1  U jeku ljeta sestra i ja nismo smjele izlaziti...   
2                   Bilo joj je zabranjeno pomagati.   
3  Kao da je ona ta koja je bespomoćnom mališanu ...   
4                   Pustio je kaput da padne na tlo.   
5  Nagomilana neaktivnost tijekom ljetnih tjedana...   
6  Bilo joj je neugodno to što je bila u svađi s ...   
7  Emily je dopustila da joj se misli odmaknu od ...   
8             Možda sam trebao Leonu napisati priču.   
9  Pretpostavljam da bismo trebali početi s čitan...   

                                    translation_User  \
0   Lijep dan poput ovoga učinio ju je nestrpljivom.   
1  Usred ljeta sestra i ja nismo smjele izlaziti ...   
2                        Zabranili su joj da pomaže.   
3  Kao da je ona bila ta koja je bespomoćnom mali...   
4                   Pustio je kaput da padne na tlo.   
5  Nagomilana neaktivnost tijekom ljetnih tjedana...   
6  Bilo joj je neugodno to što je bila u svađi s ...   
7  Emily je dopustila da joj misli odmaknu od naj...   
8       Možda sam trebala napisati Leonu neku priču.   
9  Pretpostavljam da bismo trebali početi s čitan...   

                             trans_description_GPT_q  trans_quality_GPT_n  \
0  The verb phrase 'učinio ju je nestrpljivom' co...                 0.95   
1  The verb phrase 'nismo smjele izaći' uses the ...                 0.80   
2  The verb phrase 'bilo joj je zabranjeno pomoći...                 0.90   
3  The verb phrase 'was imposing' is translated a...                 0.90   
4  The verb phrase 'let fall' is translated as 'p...                 0.95   
5  The verb phrase 'požurivala' is a past tense f...                 0.90   
6  The verb phrase 'biti u sukobu' uses the infin...                 0.80   
7  The verb phrase 'let her thoughts move away' i...                 0.90   
8  The verb phrase 'should have written' is trans...                 0.90   
9  The verb phrase 'početi čitajući ga' uses the ...                 0.70   

                          trans_description_Google_q  trans_quality_Google_n  \
0  The verb phrase 'učinio ju je nestrpljivom' us...                    0.95   
1  The verb phrase 'nismo smjele izlaziti' uses t...                    0.90   
2  The verb phrase 'bilo joj je zabranjeno pomaga...                    0.85   
3  The verb phrase 'was imposing' is translated a...                    0.70   
4  The verb phrase 'let fall' is translated as 'p...                    0.95   
5  The verb phrase 'požurivala' is used in the pa...                    0.95   
6  The verb phrase 'bila u svađi' uses the past t...                    0.90   
7  The verb phrase 'let her thoughts move away' i...                    0.80   
8  

In [ ]:
dict_responses_Croatian.to_excel('Croatian_evaluation.xlsx')

**Sample 1:** *A fine day like this **made** her **impatient**.*

*> FINNISH*

 **Human Translation:** *Tällainen upea päivä **sai** hänet **kärsimättömäksi**.*

 GPT Description:

 GPT Score:

 **GPT Translation:** *Näin kaunis päivä **teki** hänestä **kärsimättömän**.*

 GPT Description:
  
  **Lexis:** The verb phrase "teki hänestä kärsimättömän" uses "teki," the past tense of "tehdä," which means "to make." The choice of "kaunis" for "fine" is more idiomatic in Finnish."
  
  **Syntax:** The verb phrase follows Finnish sentence structure: subject + verb + object + adjective in translative case ("kärsimättömän").
  
  **Semantics:** "The verb "teki" correctly conveys the causative aspect of the English "made," aligning with Talmy's force dynamics by indicating the day's influence on her state of being.

 GPT Score: **0.95**

 **Google Translation:** *Tällainen hieno päivä **sai** hänet **kärsimättömäksi**.*

 GPT Description:

 **Lexis:** The verb phrase "sai hänet kärsimättömäksi" uses "sai," the past tense of "saada," which can mean "to make" or "to get". The word "hieno" is a direct translation of "fine" but may not capture the same idiomatic nuance."

 **Syntax:** "The verb phrase structure is correct for Finnish: subject + verb + object + adjective in translative case ("kärsimättömäksi").  

 **Semantics:** The verb "sai" is semantically accurate, indicating that the day caused a change in her, which is consistent with Talmy's force dynamics, but it may imply a slightly different nuance of causation compared to "teki."

 GPT Score: **0.9**


**Sample 2:** *In high summer, my sister an I **were not allowed out** of the house.*

*> FINNISH*

 **Human translation:** ***Emme päässeet*** *kesällä* ***ulos***.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Kesäkuumalla sisareni ja minä **emme** **saaneet** **poistua** talosta.*.*

 GPT Description:
  
  **Lexis:** The verb phrase "emme saaneet poistua" consists of the negative auxiliary verb "emme," the main verb "saaneet" in the past tense, and the infinitive "poistua." The lexis is appropriate, using "saada" to express permission.
  
  **Syntax:** The syntax follows Finnish structure with the negative auxiliary preceding the main verb.
  
  **Semantics:** Semantically, the verb phrase captures the prohibition or lack of permission, aligning with Talmy's notion of force dynamics by indicating an external force (the implied authority) restricting the agents' (the speaker and sister) actions."

 GPT Score: **0.9**

 **Google Translation:** *Keskikesällä siskoni ja minua **ei** **päästetty** **ulos** kotoa.*

 GPT Description:

 **Lexis:** "The verb phrase "ei päästetty" is composed of the negative particle "ei" and the passive past tense verb "päästetty." Lexically, "päästetty" is a valid translation for "allowed out," but it is less specific about the subject who is not allowing the action.

 **Syntax:** The syntax is correct for a Finnish passive construction.

 **Semantics:** Semantically, the phrase implies an external force preventing the action, which is in line with Talmy's force dynamics, but it lacks the explicit mention of the agents being restricted.

 GPT Score: **0.8**

**Sample 3:** *I **had** **asked** him **to be frank**.*

*> FINNISH*

 **Human Translation:** ***Olin*** *varta vasten* ***pyytänyt*** *lääkäriä* ***puhumaan kiertelemättä***.

 GPT Description:

 GPT Score:

 **GPT translation:** ***Olin*** ***pyytänyt*** *hänta* ***olemaan*** ***suorasukainen***.

 GPT Description:
  
  **Lexis & Syntax:** The verb phrase "olin pyytänyt" is composed of the auxiliary verb "olin" (the past tense of "olla," "to be") and the main verb "pyytänyt" (past active participle of "pyytää," "to ask"). This construction accurately reflects the past perfect tense used in the original English sentence, indicating an action completed in the past. The infinitive "olemaan" ("to be") is used with the adjective "suorasukainen" ("frank," "straightforward"), which is a good lexical choice for "frank." The syntax follows Finnish language conventions, where the infinitive form of the verb is used after a verb of asking or telling, and the word order is standard for Finnish.
  
  **Semantics:** Force dynamics?

 GPT Score: **0.95**

 **Google Translation:** ***Pyysin*** *häntä **olemaan** **rehellinen**.*

 GPT Description:

 **Lexis & Syntax:**  The verb phrase "pyysin" is the first person singular past tense of "pyytää," which translates to "I asked." The verb phrase is correctly formed and conveys the action of asking. However, the use of the past simple tense "Pyysin" instead of the past perfect "Olin pyytänyt" results in a slight loss of the nuance of an action that had been completed in the past. The infinitive "olemaan" is correctly used, followed by the adjective "rehellinen" ("honest"), which is a close synonym to "frank" but may carry a slightly different connotation.

 **Semantics:** Force dynamics?

 GPT Score: **0.85**

**Sample 4:** *She broke down completely and **was led** towards the drawing room.*

*> FINNISH*

 **Human Translation:** *Hän lopulta murtui kokonaan, minkä jälkeen häntä* ***lähdettiin*** *viemään saliin*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Hän romahti täysin ja **vietiin** piirustushuoneeseen.*

 GPT Description:
  
  **Lexis:** The verb "romahti" accurately captures the sense of "broke down" in terms of an emotional or physical collapse. "Vietetin" is a passive form of "viedä," meaning "to lead" or "to take," which is appropriate in this context.
  
  **Syntax:** "The verb is in the past tense, third-person singular, which aligns with the subject "hän" ("she"). The verb "vietiin" is in the passive voice, past tense, indicating that the subject is being acted upon.
  
  **Semantics:** The verb "romahti" conveys a sudden and complete loss of control or function, aligning with Talmy's notion of force dynamics where the subject experiences an inward force overpowering her structural integrity. The verb "vietiin" implies that the subject is being guided by an external force, consistent with Talmy's force dynamics, where an agent exerts force on the subject to move her towards a destination.

 GPT Score: **0.95**

 **Google Translation:** *Hän murtui täysin ja hänet **johdettiin** olohuoneeseen.*

 GPT Description:

 **Lexis:** "The verb "murtui" is a good lexical choice for "broke down," suggesting a breakdown or collapse. "Johdettiin" is the passive form of "johtaa," which can mean "to lead" but is less commonly used in the context of guiding a person to a room."

 **Syntax:** The verb "murtui" is in the past tense, third-person singular, matching the subject "hän." The verb "johtaa" is correctly used in the passive voice, past tense.

 **Semantics:** "The verb "murtui" conveys a sense of breaking or giving way under pressure, which is semantically similar to the English phrase "broke down" and aligns with the concept of force dynamics. "The verb "johdettiin" implies being guided or directed, which fits the context but may not be as idiomatic in Finnish for this particular scenario.

 GPT Score: **0.85**

**Sample 5:** *She **was forbidden to help**.*

*> FINNISH*

 **Human Translation:** ***Auttaa*** *hän* ***ei saanut***.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Hänen* ***auttamisensa kiellettiin***.

 GPT Description:
  
  **Lexis:** The verb phrase "auttamisensa kiellettiin" uses a possessive structure ("hänen auttamisensa") which is not as common in Finnish as in English. The verb "kiellettiin" is in the passive past tense."
  
  **Syntax:** The verb phrase follows Finnish syntax with the verb in the passive form, reflecting the subject's lack of agency. The noun "auttamisensa" is in the genitive case, showing possession.
  
  **Semantics:** The verb phrase captures the meaning of prohibition and the past tense. However, the use of the possessive structure slightly shifts the focus onto the act of helping as something that belongs to her, which is not a direct match with the English semantics.

 GPT Score: **0.8**

 **Google Translation:** *Häntä* ***kiellettiin*** ***auttamasta***.

 GPT Description:

 **Lexis:** "he verb phrase "kiellettiin auttamasta" uses the standard Finnish construction for expressing prohibition. The verb "kiellettiin" is in the passive past tense, and "auttamasta" is in the elative case, indicating the action from which she is being prohibited."    

 **Syntax:** The verb phrase is syntactically standard for Finnish. The use of the passive voice "kiellettiin" correctly indicates that the action was done to her, and the elative case "auttamasta" is typical for expressing prohibition from an action."

 **Semantics:** The verb phrase accurately conveys the meaning of being prohibited from an action, aligning well with Talmy's notion of force dynamics, where an external force (the prohibition) acts upon the subject, preventing her from performing the action (helping).

 GPT Score: **0.95**

**Sample 6:** *As though she were the one who **was imposing** unpleasant **ordeals** on a helpless little fellow.*

*> FINNISH*

 **Human Translation:** *Ikään kuin juuri Briony* *olisiki ollut se, joka* ***yritti pakotta*** *avutonta pikkumiestä* ***epämieluisiin koettelemuksiin***.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Kuin hän olisi se, joka* ***saattoi*** *avuttoman pikkumiehen* *epämiellyttävien* ***koettelemusten*** ***eteen***.

 GPT Description:
  
  **Lexis & Syntax:** In the GPT translation, the verb phrase "saattoi ... eteen" is used, which translates to "led ... in front of" or "brought ... before." The lexis here is slightly more nuanced than a direct translation, implying a guiding or leading action. The syntax is standard for Finnish, with "saattoi" being the past tense form of the verb "saattaa," and "eteen" serving as a postposition that requires the genitive case "koettelemusten."
  
  **Semantics:** Semantically, the verb phrase suggests an action of causing someone to face something, which aligns with Talmy's notion of force dynamics, where the subject (she) is the force vector causing the action upon the object (the little fellow)."

 GPT Score: **0.9**

 **Google Translation:** *Ikään kuin hän olisi se, joka* ***asetti*** ***epämiellyttäviä*** ***koettelemuksia*** *avuttomalle pienelle miehelle*.

 GPT Description:

 **Lexis & Syntax:** The Google translation uses the verb phrase "asetti ... koettelemuksia," which directly translates to "placed ... ordeals." The lexis is straightforward and clear, with "asetti" being the past tense of "asettaa," meaning "to place" or "to set." The syntax is correct, with the verb preceding the object as is typical in Finnish.

 **Semantics:** Semantically, the verb phrase "asetti" indicates the act of placing or setting something, which is a slightly more static action than "saattoi ... eteen" and does not convey the same sense of motion or guidance. However, it still fits within the framework of force dynamics, with the subject exerting control over the object."

 GPT Score: **0.85**

**Sample 7:** *He **let** the greatcoat **fall** on the ground.*

*> FINNISH*

 **Human Translation:** *Hän* ***päästi*** *manttelinsa* ***putoamaan*** *maahan*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Hän* ***antoi*** *suuren takin* ***pudota*** *maahan*.

 GPT Description:
  
  **Lexis:** The verb phrase "antoi ... pudota" accurately captures the permissive aspect of "let fall." The noun "suuren takin" is a more explicit translation of "greatcoat" than the Google translation.
  
  **Syntax:** The verb phrase structure follows Finnish syntax with the main verb "antoi" followed by an infinitive "pudota." The word order is subject-verb-object, which is standard in Finnish."
  
  **Semantics:** The verb phrase conveys the causative-permissive force dynamic where the subject ("hän") allows an event ("pudota") to happen without direct causation, aligning with Talmy's notion of force dynamics.

 GPT Score: **0.9**

 **Google Translation:** *Hän* ***antoi*** *suuren takin* ***pudota*** *maahan*.

 GPT Description:

 **Lexis:** The verb phrase "antoi ... pudota" is a correct translation of "let fall," but the noun "takin" is less specific than "greatcoat."      

 **Syntax:** The structure "antoi ... pudota" follows the Finnish convention of expressing "let" with the verb "antaa" followed by an infinitive, maintaining proper syntax.    

 **Semantics:** The verb phrase "antoi ... pudota" correctly conveys the permissive action of allowing something to fall, which is consistent with the semantics of the original English sentence and Talmy's force dynamics.

 GPT Score: **0.8**

**Sample 8:** *The accumulated inactivity of the summer weeks since finals also **hurried** her **along**.*

*> FINNISH*

 **Human Translation:** *Kai hän* ***kiiruhti*** *askeliaan siksikin, että oli ollut toimettomana kaikki nämä loppukokeiden jälkeiset kesäiset viikot*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Kesäviikkojen kasaantunut toimettomuus finaalien jälkeen* ***kiirehti*** *häntä myös* ***eteenpäin***.

 GPT Description:
  
  **Lexis:** The verb phrase "kiirehti häntä myös eteenpäin" uses "kiirehti" as the main verb, which translates to "hurried" and is lexically accurate. The verb is in the past tense, third-person singular form, which aligns with the subject "toimettomuus" ("inactivity"). The adverb "myös" ("also") is correctly placed before the verb phrase to mirror the adverbial placement in the original sentence.
  
  **Syntax:** The verb phrase follows Finnish syntax rules, with the verb preceding the object "häntä" ("her") and the adverbial "eteenpäin" ("along") at the end. This word order is typical for Finnish, where the verb generally comes before the object.    
  
  **Semantics:** Semantically, the verb "kiirehti" conveys a sense of urgency, which is in line with the original English verb "hurried." The use of "eteenpäin" ("along") captures the directional aspect of "hurried her along." In terms of force dynamics, the verb phrase suggests that the "inactivity" exerts a force that propels the subject forward, which is consistent with the original.

 GPT Score: **0.9**

 **Google Translation:** *Myös finaalien jälkeen kertynyt kesäviikkojen passiivisuus* ***kiihdytti*** *häntä* ***mukaan***.

 GPT Description:

 **Lexis:** The verb phrase "kiihdytti häntä mukaan" uses "kiihdytti," which is a form of "kiihdyttää," meaning "to accelerate." This choice of verb introduces a slightly different nuance than "hurried," suggesting a more intense action. The adverb "mukaan" ("along") is used to indicate inclusion or participation, which is not a direct translation of "along" in the original sentence.    

 **Syntax:** The syntax of "kiihdytti häntä mukaan" follows Finnish structure, with the verb "kiihdytti" preceding the object "häntä" and the adverb "mukaan" following the verb. The placement of "myös" ("also") at the beginning of the sentence is syntactically correct in Finnish."     

 **Semantics:** Semantically, "kiihdytti" implies a force that causes an increase in speed or intensity, which is a more forceful action than "hurried." The use of "mukaan" does not fully capture the directional movement implied by "along" in the original text. In terms of force dynamics, the translation suggests a stronger compulsion or push exerted on the subject, which may not align perfectly with the original sense of being hurried along by inactivity.

 GPT Score: **0.75**

**Sample 9:** *Being at odds with her father about anything at all **made** her **uncomfortable**.*

*> FINNISH*

 **Human Translation:** *Hänestä* ***tuntui*** *kovin* ***epämukavalta*** *joutua ylipäänsä minkäänlaiseen hakaukseen isän kanssa*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Olla eri mieltä isänsä kanssa mistä tahansa asiasta* ***sai*** *hänet* ***tuntemaan*** ***olonsa*** ***epämukavaksi***.

 GPT Description:
  
  **Lexis & Syntax:** The verb phrase "sai hänet tuntemaan olonsa epämukavaksi" consists of the verb "sai" ("made"), which is in the past tense and indicative mood, the pronoun "hänet" ("her") as the direct object, and the infinitive verb phrase "tuntemaan olonsa epämukavaksi" ("to feel uncomfortable"). The lexis is appropriate, using standard Finnish verb forms and pronouns. The syntax follows Finnish sentence structure, with the verb preceding its object.   
  
  **Semantics:** Semantically, the verb "sai" indicates a causative force dynamic, where the subject's state is changed by the action of being at odds.

 GPT Score: **0.9**

 **Google Translation:** *Ristiriidat isänsä kanssa ylipäänsä mistä tahansa* ***sai*** *hänet* ***epämukavaksi***.

 GPT Description:

 **Lexis & Syntax:** The verb phrase "sai hänet epämukavaksi" includes the verb "sai" ("made"), the pronoun "hänet" ("her") as the direct object, and the adjective "epämukavaksi" ("uncomfortable") in the translative case, indicating a change of state. The lexis is slightly less precise due to the omission of the verb "to feel," which affects the nuance of the sentence. The syntax is correct but less explicit without the verb "to feel."

 **Semantics:** Semantically, the verb "sai" still conveys a causative force dynamic, but the translation is less detailed in expressing the internal experience of discomfort.

 GPT Score: **0.8**

**Sample 10:** *Emily **let** her thoughts **move away** from her eldest daughter.*

*> FINNISH*

 **Human Translation:** *Emily* ***antoi*** *ajatustensa* ***siirtyä*** ***pois*** *vanhemmasta tyttärestä*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Emily* ***salli*** *ajatustensa* ***kulkeutua*** ***pois*** *vanhimman tyttärensä luota*.

 GPT Description:
  
  **Lexis & Syntax:** In the translation provided by GPT, the verb phrase "salli ajatustensa kulkeutua" is used. "Salli" is the third person singular past tense form of "sallia," which means "to allow" or "to let." The verb "kulkeutua" is an intransitive verb in the infinitive form, which means "to drift" or "to move." The verb phrase is syntactically correct, following Finnish sentence structure where the main verb "salli" is followed by an object in its possessive form "ajatustensa" ("her thoughts") and a secondary verb in the infinitive form "kulkeutua."
  
  **Semantics:** Semantically, the verb phrase captures the sense of allowing a process to happen, which aligns with Talmy's notion of force dynamics, indicating a lack of force or resistance in the action of the thoughts moving.

 GPT Score: **0.9**

 **Google Translation:** *Emily* ***antoi*** *ajatustensa* ***siirtyä*** ***pois*** *vanhimmasta tyttärestään.*

 GPT Description:

 **Lexis & Syntax:** The Google translation uses the verb phrase "antoi ajatustensa siirtyä." "Antoi" is the third person singular past tense form of "antaa," which means "to give" but can also be used in the sense of "to let." The verb "siirtyä" is an intransitive verb in the infinitive form, meaning "to move" or "to shift." The syntax is standard for Finnish, with the main verb "antoi" followed by the object in possessive form "ajatustensa" and the secondary verb "siirtyä."

 **Semantics:** Semantically, the verb phrase "antoi siirtyä" is slightly less idiomatic in this context but still conveys the intended meaning of permitting an action to take place, which is consistent with Talmy's Force Dynamics, suggesting a permissive action without direct involvement.

 GPT Score: **0.85**

**Sample 11:** *Every Ares mission **requires** three years of pre-supplies.*

*> FINNISH*

 **Human Translation:** *Ares-lento* ***vaatii*** *kolmen vuoden tarvikkeet*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Jokainen Ares-tehtävä* ***edellyttää*** *kolmen vuoden ennakkotarvikkeita*.

 GPT Description:
  
  **Lexis & Syntax:** The verb phrase "edellyttää kolmen vuoden ennakkotarvikkeita" uses "edellyttää" as the main verb, which translates to "requires." The verb is in present tense, indicative mood, and active voice, which is syntactically correct for the given context. The lexis is appropriate, with "edellyttää" being a common verb for expressing necessity.
  
  **Semantics:** Semantically, the verb phrase conveys the necessity of having supplies in advance, aligning with Talmy's notion of Force Dynamics by indicating that the mission's success is contingent upon the action of securing supplies beforehand.

 GPT Score: **0.95**

 **Google Translation:** *Jokainen Ares-tehtävä* ***vaatii*** *kolmen vuoden esitoimituksia*.

 GPT Description:

 **Lexis & Syntax:** The verb phrase "vaatii kolmen vuoden esitoimituksia" uses "vaatii" as the main verb, which also translates to "requires." The verb is in present tense, indicative mood, and active voice. The lexis "vaatii" is a direct translation of "requires" and is commonly used in Finnish. However, "esitoimituksia" is a less common term for "pre-supplies" and might not be as idiomatic as "ennakkotarvikkeita."

 **Semantics:** Semantically, the verb phrase correctly conveys the requirement aspect, but the choice of "esitoimituksia" could slightly affect the naturalness of the phrase in Finnish.

 GPT Score: **0.9**

**Sample 12:** *Mars **would be governed** by maritime law.*

*> FINNISH*

 **Human Translation:** *Mars* ***on*** *merilakien* ***alaista***.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Marsia* ***hallittaisiin*** *merioikeuden mukaisesti*.

 GPT Description:
  
   **Lexis:** "The verb phrase "hallittaisiin" is a passive conditional form of "hallita," meaning "to govern" or "to rule." The adverbial phrase "merioikeuden mukaisesti" translates to "according to maritime law," where "merioikeuden" is the genitive form of "merioikeus" ("maritime law") and "mukaisesti" means "according to."
  
  **Syntax:** The verb phrase is in the passive voice, which is appropriate for the impersonal nature of the original sentence. The conditional mood is used to express the hypothetical aspect of the governance of Mars. The adverbial phrase follows the verb, which is typical for Finnish sentence structure.
  
  **Semantics:** The verb phrase captures the hypothetical and regulatory aspects of the original sentence. In terms of force dynamics, the verb "hallittaisiin" implies that Mars would be subject to an external force, which is maritime law.

 GPT Score: **0.95**

 **Google Translation:** *Marsia* ***säännelisi*** *merilaki*.

 GPT Description:

  **Lexis:** The verb phrase "säännelisi" is a conditional form of "säännellä," meaning "to regulate" or "to control." The noun "merilaki" translates directly to "maritime law."  
  
  **Syntax:** The verb phrase is in the active voice, which is less common for expressing governance in Finnish when the agent is not specified. The conditional mood is correctly used to express the hypothetical nature of the statement.
  
  **Semantics:** The verb phrase "säännelisi" implies regulation, which is a slightly different nuance than "governed." In terms of force dynamics, "säännelisi" suggests that maritime law would actively exert control over Mars, which is a reasonable interpretation but slightly different from the original.

 GPT Score: **0.85**

**Sample 13:** *You **ought to call** the police.*

*> FINNISH*

 **Human Translation:** ***Soita*** *poliisille*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Sinun* ***tulisi soittaa*** *poliisille*.

 GPT Description:
  
  **Lexis & Syntax:** The verb phrase "tulisi soittaa" is composed of the modal auxiliary "tulisi" (a conditional form of "tulla," meaning "should" or "ought to") and the main verb "soittaa" ("to call"). The syntax follows standard Finnish structure for expressing a suggestion or obligation, with the modal auxiliary preceding the infinitive verb.
  
  **Semantics:** Semantically, the verb phrase conveys an obligation or recommendation, aligning with Talmy's notion of force dynamics by suggesting that an external force (the speaker's opinion or societal expectation) is acting upon the subject to perform the action."  

 GPT Score: **0.95**

 **Google Translation:** *Sinun* ***pitäisi soittaa*** *poliisille*.

 GPT Description:

 **Lexis & Syntax:** The verb phrase "pitäisi soittaa" consists of the modal auxiliary "pitäisi" (a conditional form of "pitää," also meaning "should" or "ought to") and the main verb "soittaa" ("to call"). The syntax is identical to the GPT translation, with the modal auxiliary coming before the infinitive verb.

 **Semantics:** Semantically, this verb phrase also suggests an obligation or recommendation, and similarly reflects force dynamics by implying that an external force influences the subject's potential action."

 GPT Score: **0.95**

**Sample 14:** *Sounds as if you **don't want** it.*

*> FINNISH*

 **Human Translation:** ***Eikö kiinnosta?***

 GPT Description:

 GPT Score:

 **GPT Translation:** *Vaikuttaa siltä,* ***ettet***  ***halua*** *sitä*.

 GPT Description:
  
   **Lexis:** "The verb phrase "vaikuttaa siltä" is a standard Finnish construction that corresponds to "sounds as if" in English. The verb "halua" is the correct form of "to want" in the negative for the second person singular.
  
  **Syntax:** "The verb phrase follows Finnish sentence structure: the main verb "vaikuttaa" is in initial position, followed by "siltä", which introduces the subordinate clause. The negation is correctly formed with "et" preceding the verb "halua."    
  
  **Semantics:** The verb "vaikuttaa" conveys a sense of impression or seeming, which aligns with Talmy's notion of force dynamics as it implies a lack of intention or desire on the part of the subject.

 GPT Score: **0.95**

 **Google Translation:** *Kuulostaa siltä, että* ***ette halua*** *sitä*.

 GPT Description:

  **Lexis:** The verb phrase "kuulostaa siltä" is a direct translation of "sounds as if." The verb "halua" is correctly used for "to want," but the form "ette" is the second person plural or formal "you, which may not match the informal singular "you" in the original.     
  
  **Syntax:** "The structure "kuulostaa siltä" is syntactically correct in Finnish. The placement of "ette," the plural or formal "you," before "halua" is the standard negation structure in Finnish.    
  
  **Semantics:** The verb "kuulostaa" correctly captures the auditory implication of "sounds," but the use of "ette" introduces a formality or plurality that may not be present in the original, potentially altering the force dynamics by addressing a group or a single individual formally.

 GPT Score: **0.85**

**Sample 15:** ***Let's release*** *the photo when we detail the rescue operation.*

*> FINNISH*

 **Human Translation:** ***Julkaisemme*** *kuvan pelastustiedotteessa*.

 GPT Description:

 GPT Score:

 **GPT Translation:** ***Julkaiskaame*** kuva, kun käymme pelastusoperaation yksityiskohdat läpi*.

 GPT Description:
  
  **Lexis & Syntax & Semantics:** The verb phrase "julkaiskaamme kuva" uses the cohortative form "julkaiskaamme," which is a first-person plural imperative in Finnish, suggesting a collective action. The verb "käymme ... läpi" combines "käymme," the first-person plural of "käydä" ("to go"), with "läpi" ("through"), forming a phrasal verb that means "to go through" or "to detail." This verb phrase captures the iterative aspect of detailing an operation.

 GPT Score: **0.95**

 **Google Translation:** ***Julkaistaan*** *kuva, kun kerromme pelastusoperaation yksityiskohtaisesti*.

 GPT Description:

 **Lexis & Syntax & Semantics:** "The verb phrase "julkaistaan kuva" uses the passive form "julkaistaan," which is a third-person singular present passive, implying an impersonal or collective action. The verb "kerromme" is the first-person plural present of "kertoa" ("to tell"), and "yksityiskohtaisesti" is an adverb meaning "in detail." The verb phrase "kerromme pelastusoperaation yksityiskohtaisesti" is a direct way of expressing the act of detailing.

 GPT Score: **0.9**

**Sample 16:** *Perhaps I **should have written** Leon a story.*

*> FINNISH*

 **Human Translation:** ***Miksen kirjoittanut*** *Leonille tarinaa?*

 GPT Description:

 GPT Score:

 **GPT Translation:** *Ehkä minun* ***olisi pitänyt kirjoittaa*** *Leonille tarina*.

 GPT Description:
  
   **Lexis:** The verb phrase "olisi pitänyt kirjoittaa" accurately captures the modal auxiliary "should have" and the main verb "written."    
  
  **Syntax:** The verb phrase follows Finnish syntax by placing the modal auxiliary "olisi" before the past participle "pitänyt" and the infinitive "kirjoittaa."
  
  **Semantics:** The verb phrase conveys the sense of a missed obligation or opportunity, aligning with Talmy's notion of force dynamics by expressing a potential force (writing a story) that was not actualised.

 GPT Score: **1.0**

 **Google Translation:** *Ehkä minun* ***olisi pitänyt kirjoittaa*** *Leonille tarina*.

 GPT Description:

  **Lexis:** The verb phrase "olisi pitänyt kirjoittaa" is lexically identical to the GPT translation, reflecting the same modal and main verb."   
  
  **Syntax:** The syntax is also identical, with the modal auxiliary and past participle preceding the infinitive form of the verb, which is standard in Finnish.
  
  **Semantics:** Semantically, the Google translation matches the GPT translation, effectively conveying the same missed obligation or opportunity and adhering to the principles of force dynamics.

 GPT Score: **1.0**

**Sample 17:** *I suppose we **should start by reading** it.*

*> FINNISH*

 **Human Translation:** ***Luetaan*** *se näytelmä*.

 GPT Description:

 GPT Score:

 **GPT Translation:** *Oletan, että meidän* ***tulisi aloittaa lukemisesta***.

 GPT Description:
  
   **Lexis:** The verb phrase "tulisi aloittaa" accurately captures the modal necessity and the infinitive form of "start," which is appropriate for Finnish. "Lukemisesta" is the elative case of "lukeminen," indicating the action "reading" as the starting point.   
  
  **Syntax:** The verb phrase follows Finnish syntax, with the modal verb "tulisi" ("should") preceding the infinitive "aloittaa" ("to start"). The use of the elative case "lukemisesta" is syntactically correct for expressing the action by which something should start.    
  
  **Semantics:** The verb phrase conveys the same sense of obligation and initiation as the English source. In terms of force dynamics, the agent (we) is being subjected to a gentle force or obligation to initiate an action (reading).  

 GPT Score: **0.95**

 **Google Translation:** *Luulen, että meidän* ***pitäisi aloittaa lukemalla*** *se*.

 GPT Description:

  **Lexis:** The verb phrase "pitäisi aloittaa" is a correct translation for "should start." "Lukemalla" is the instructive case of "lukeminen," which is used for the means by which something is done.
  
  **Syntax:** The verb phrase "meidän pitäisi aloittaa" follows Finnish syntax, with the possessive pronoun "meidän" ("our") followed by the modal verb "pitäisi" ("should") and the infinitive "aloittaa" ("to start"). The use of "lukemalla" ("by reading") is syntactically correct but less common in Finnish for expressing starting an action.  
  
  **Semantics:** The verb phrase semantically matches the English source, indicating an obligation to initiate an action (reading). However, the use of the instructive case "lukemalla" slightly shifts the focus to the means of initiation rather than the action itself, which is a subtle difference in force dynamics.

 GPT Score: **0.9**